cdap-api/src/main/java/co/cask/cdap/api/data/batch/Input.java

```diff
@@ -74,41 +74,38 @@ public Input alias(String alias) {
    *
    * @param datasetName the name of the input dataset
    */
-  public static Input ofDataset(String datasetName) {
+  public static DatasetInput ofDataset(String datasetName) {
```

Comment:
Users are expected to use the API as `Input.ofDataset().fromNamespace()`.
Without this change, a type conversion is required as `((DatasetInput) Input.ofDataset()).fromNamespace()`